torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1811,6 +1833,19 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
            decorators=[_wrap_maybe_warns("floor_divide is deprecated, and will be removed")],
            supports_autograd=False,
            ),
+    OpInfo('linalg.matrix_power',
+           op=torch.linalg.matrix_power,
+           dtypes=floating_and_complex_types(),
+           test_inplace_grad=False,
+           supports_tensor_out=True,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+           sample_inputs_func=sample_inputs_linalg_matrix_power,
+           aten_name='linalg_matrix_power',
+           skips=(
+               # RuntimeError: input->type()->kind() == TypeKind::OptionalTypeINTERNAL ASSERT FAILED
```

Comment:
Would you elaborate on this? If this issue isn't addressed in this PR let's file an issue and post its url in this comment.

Reply:
Removed the skip, tests seems to pass now.